packages/editor/src/components/blocks/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,36 @@
+import type { Checkbox, Prettify } from '@axonivy/form-editor-protocol';
+import type { ComponentConfig, UiComponentProps } from '../../../types/config';
+import './Checkbox.css';
+import { baseComponentFields, defaultBaseComponent } from '../base';
+import IconSvg from './Checkbox.svg?react';
+import { BasicCheckbox } from '@axonivy/ui-components';
+
+type CheckboxProps = Prettify<Checkbox>;
+
+export const defaultCheckboxProps: Checkbox = {
+  label: 'Label',
+  selected: false,
+  ...defaultBaseComponent
+} as const;
+
+export const CheckboxComponent: ComponentConfig<CheckboxProps> = {
+  name: 'Checkbox',
+  category: 'Elements',
+  subcategory: 'Interactions',
+  icon: <IconSvg />,
+  description: 'A selectable boolean checkbox',
+  defaultProps: defaultCheckboxProps,
+  render: props => <UiInput {...props} />,
+  create: ({ label, value, ...defaultProps }) => ({ ...defaultCheckboxProps, label, value, ...defaultProps }),
+  fields: {
+    label: { subsection: 'General', label: 'Label', type: 'text' },
+    selected: { subsection: 'General', label: 'Selected', type: 'checkbox' },
```

Comment:
Makes sense, should a validation happen here somewhere, which checks that this string is really only "true" or "false"?

Reply:
what do you mean with really only "true" or "false"? The value of this selected property will be something like "#{data.accepted}".

Reply:
Yes, I am aware of that, I meant that you check in the backend that the actuall value of data.accepted is true or false (a Boolean) and not any other string. but I just noticed that if data.accepted is any other string the checkbox is simply deselected, i.e. false. 
Initially I deliberately set the selected attribute in the backend to Boolean to ensure that the value of the component can only ever be true or false.
![grafik](https://github.com/user-attachments/assets/193ef38f-356b-4366-a34c-be40be6cec20)

Reply:
Ah yes, as you mentioned seems no so problematic. But yes some sort of validation is planed. E.g. that a data field is really existing, and we can of course also think about warning if the type of the data attribute is not something we expect (e.g. in this case a boolean)

Reply:
Perfect, is there already a validation story? I guess this one: https://1ivy.atlassian.net/browse/XIVY-13626
Otherwise I would document this explicit point there.